packages/gatsby-cli/src/reporter/reporter.ts

```diff
@@ -44,7 +44,7 @@ class Reporter {
    * the reporter to extend the internal error map
    */
 
-  setErrorMap = (entry: Record<ErrorId, IErrorMapEntry>): void => {
+  setErrorMap = (entry: Record<string, IErrorMapEntry>): void => {
```

Comment:
Might be an idea to add a sentence to the jsdoc emphasising the fact that the IDs should be different from built-in ones.